core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -115,7 +117,17 @@ class HistoryServer(
   }
 
   def getSparkUI(appKey: String): Option[SparkUI] = {
-    Option(appCache.get(appKey))
```

Comment:
`appCache.getIfPresent` returns null if there is no cached value for the appKey. But `appCache.get` will try to obtain that value from a `CacheLoader`, cache it and return it. So I think we still need use `appCache.get` here and handle the exception.